src/main/java/carbon/task/Todo.java

```diff
@@ -0,0 +1,67 @@
+package carbon.task;
+
+import carbon.error.CarbonException;
+import carbon.error.InvalidParamException;
+
+/**
+ * Enapsulates information regarding Todos.
+ */
+public class Todo extends Task {
+    private static final int TYPEKEY = Task.getTypeKey(Task.Type.TODO);
+
+    private Todo(String name, Boolean isDone) {
+        super(name, isDone);
+    }
+
+    private static String extractName(String input) throws CarbonException {
+        int len = input.length();
+        int requiredLen = "todo ".length();
```

Comment:
Just want to commend you on this point here and overall in the code - I really like how you make it a point not to use any magic numbers and not take shortcuts. Well done!